---
title: "Shared ASV Analysis for MARCH 16S Microbiome (12902)"
author: "Tiange Liu"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float:
      collapsed: false
      smooth_scroll: false
knit: (function(input, ...) {
    rmarkdown::render(
      input,
      output_dir = "Z:/DAC_AnalysisProjects/10_Analysis_Proposals/EC0556A/USERS/TiangeLiu/OUTPUT/Results",  
    )
  })
---

```{r set up, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r include=FALSE}
source("Z:/DAC_AnalysisProjects/10_Analysis_Proposals/EC0556A/USERS/TiangeLiu/ANALYSIS/Codes/OIF_Function.r")
path <- paste("Z:/DAC_AnalysisProjects/10_Analysis_Proposals/EC0556A/USERS/TiangeLiu/OUTPUT/")
```

# Read in covariates 
```{r}
master_dt <- read.csv("Z:/DAC_AnalysisProjects/10_Analysis_Proposals/EC0556A/DATA/DERIVED/OIF_master_SQL_072823_v3.csv")

# re-label some variables
master_dt$birth_delivm <- as.factor(master_dt$birth_delivm)
master_dt$birth_delivm <- factor(master_dt$birth_delivm, levels=c("-5", "1","2"), labels=c("Missing", "Vaginal delivery", "Cesarean delivery"))

master_dt$med_antib_pn <- as.factor(master_dt$med_antib_pn)
master_dt$med_antib_pn <- factor(master_dt$med_antib_pn, levels=c("-8", "0", "1"), labels=c("Missing", "No antibiotics", "Took antibiotics"))
```


# Read in paired IDs
```{r}
march <- read.csv("Z:/DAC_AnalysisProjects/10_Analysis_Proposals/EC0556A/DATA/DERIVED/12902/MARCH_paired_54.csv")

march$participant_type <- as.factor(march$m16s_part_type)
march$participant_type <- factor(march$participant_type, levels=c("1", "2"), labels=c("biological mother", "child"))

march$body_site <- as.factor(march$m16s_spec_type)
march$body_site <- factor(march$m16s_spec_type, levels=c("1", "7"), labels=c("stool", "vaginal"))

march$paired_ID <- substr(march$ParticipantID, 1, 9)
```

# Read in breastfeeding variable
```{r message=FALSE}
# Add breastfeeding variables measured at the time of sample collection
bf_dt <- read_tsv("Z:/DAC_AnalysisProjects/10_Analysis_Proposals/EC0556A/DATA/DERIVED/20230518/01.CohortMerge/01.metadata/submitted_participants_0556a.tsv")


table(bf_dt$child_breastfed, bf_dt$infant_diet)
# breastfed: 1=yes, 2=no
# diet:1, Breastmilk| 2, Infant formula | 3, Mixed breastmilk and formula |4, Breastmilk and solids| 5, Formula and solids| 6, Breastmilk, formula and solids|7, Other| -7,Declined | -8,Donâ€™t know | -9,Missing 

bf_dt <- bf_dt %>%
  select('sample-id', child_breastfed, infant_diet)
march <- merge(march, bf_dt, by.x=c("sample.id"), by.y="sample-id")
table(march[march$participant_type=="child",]$child_breastfed, march[march$participant_type=="child",]$infant_diet)

# generate a new variable
march$feeding <- ifelse(march$participant_type=="biological mother", -6, 
                        ifelse(march$infant_diet==1, 1, ifelse(march$infant_diet==2 & march$child_breastfed==2, 2, 3)))
table(march[march$participant_type=="child",]$feeding)  #34, 6, 14
march$feeding <- as.factor(march$feeding)
march$feeding <- factor(march$feeding, levels=c("-6", "1", "2", "3"), labels=c("NA", "Exclusive breastmilk", "Exclusive formula", "Mixed"))


march$bf <- ifelse(march$participant_type=="biological mother", -6,
                   ifelse(march$feeding=="Exclusive formula", 2, 1))
table(march[march$participant_type=="child",]$bf)  #48, 6
march$bf <- as.factor(march$bf)
march$bf <- factor(march$bf, levels=c("-6", "1", "2"), labels=c("NA", "Yes", "No"))
```


# Compare sequencing depth by participant type and sample type
```{r echo=FALSE, fig.height=5, fig.width=9}
p1 <- march %>%
  mutate(group=ifelse(participant_type=="child", "infant fecal", ifelse(participant_type=="biological mother" & body_site=="stool", "maternal fecal", "maternal vaginal"))) %>%
  ggpaired(x = "group", y = "m16s_slv138_97_depth",
         color = "group", line.color = "gray", line.size = 0.4,
         palette = "jco")+  
  stat_compare_means(comparisons = list(c("infant fecal", "maternal fecal"), c("infant fecal", "maternal vaginal"), c("maternal fecal", "maternal vaginal")), paired = TRUE) + # Add pairwise comparisons p-value, wilcoxon 
    scale_y_continuous(labels = scales::comma)

p2 <- march %>%
  mutate(group=ifelse(participant_type=="child", "infant fecal", ifelse(participant_type=="biological mother" & body_site=="stool", "maternal fecal", "maternal vaginal"))) %>%
  ggpaired(x = "group", y = "m16s_slv138_97_depth",
         color = "group", line.color = "gray", line.size = 0.4,
         palette = "jco")+  
  stat_compare_means(comparisons = list(c("infant fecal", "maternal fecal"), c("infant fecal", "maternal vaginal"), c("maternal fecal", "maternal vaginal")), method="t.test", paired = TRUE) + # Add pairwise comparisons p-value, t test
      scale_y_continuous(labels = scales::comma)

ggarrange(p1, p2, labels = c("Wilcoxon", "t tests"), nrow=1)
```

## Compare sequencing depth of vaginal samples by trimester
```{r echo=FALSE, fig.height=5, fig.width=9}
p1 <- march[march$body_site=="vaginal",] %>%
  ggboxplot(x = "m16s_coll_tmst", y = "m16s_slv138_97_depth",
         color = "m16s_coll_tmst", line.color = "gray", line.size = 0.4,
         palette = "jco", add = "jitter")+  
  stat_compare_means(comparisons = list(c("1", "2"), c("1", "3"), c("2", "3"))) + 
  scale_y_continuous(labels = scales::comma) 

p2 <- march[march$body_site=="vaginal",] %>%
  ggboxplot(x = "m16s_coll_tmst", y = "m16s_slv138_97_depth",
         color = "m16s_coll_tmst", line.color = "gray", line.size = 0.4,
         palette = "jco", add = "jitter")+  
  stat_compare_means(comparisons = list(c("1", "2"), c("1", "3"), c("2", "3")), method="t.test") + 
  scale_y_continuous(labels = scales::comma) 

ggarrange(p1, p2, labels = c("Wilcoxon", "t tests"), nrow=1)
```


# Read in ASV table and modify
```{r message=FALSE, warning=FALSE}
asv_table <- read_tsv("Z:/DAC_AnalysisProjects/10_Analysis_Proposals/EC0556A/DATA/DERIVED/20230518/01.CohortMerge/03.cohort_asv_data/12902/12902_asv_table_classic.tsv")
  # this is newer asv table Justine created in May 2023

asv_table <- as.data.frame(t(asv_table)) 
  # rows are samples, columns are taxa
colnames(asv_table) <- asv_table[1, ]
  # assign colnames as ASV id
asv_table <- asv_table[-(1:2), ] # remove unnecessary rows

# only keep paired samples
asv_table_paired <- asv_table %>% filter(row.names(asv_table) %in% as.list(march$sample.id))
row_names <- row.names(asv_table_paired)

str(asv_table_paired[1:5, 1:5]) # counts are not integer now
asv_table_paired <- as.data.frame(lapply(asv_table_paired, as.integer)) 
  # transform counts to integer

row.names(asv_table_paired) <- row_names
```

## Extract ASV taxanomy and modify names
```{r}
asv_name <- fun_asv_name(asv_table)
```


# Determine sharing of each ASV for each infant 
```{r}
# Split into three ASV tables
infant_id <- filter(march, participant_type=="child")
asv_infant <- asv_table_paired %>% filter(row.names(asv_table_paired) %in% as.list(infant_id$sample.id))
asv_infant <- asv_infant[order(row.names(asv_infant)), ]

vagina_id <- filter(march, body_site=="vaginal")
asv_vagina <- asv_table_paired %>% filter(row.names(asv_table_paired) %in% as.list(vagina_id$sample.id))
asv_vagina <- asv_vagina[order(row.names(asv_vagina)), ]

stool_id <- filter(march, participant_type=="biological mother" & body_site=="stool")
asv_stool <- asv_table_paired %>% filter(row.names(asv_table_paired) %in% as.list(stool_id$sample.id))
asv_stool <- asv_stool[order(row.names(asv_stool)), ]

## Definition 1: When present is defined as count>=1
share_res <- asv_infant

for(i in 1:ncol(share_res)){
  share_res[, i] <- ifelse(((asv_infant[,i]>0) & (asv_vagina[,i]>0) & (asv_stool[,i]>0)), 3, # shared with both maternal vagina and stool
                           ifelse(((asv_infant[,i]>0) & (asv_vagina[,i]>0) & (asv_stool[,i]==0)), 1, # shared with maternal vagina only
                                  ifelse(((asv_infant[,i]>0) & (asv_vagina[,i]==0) & (asv_stool[,i]>0)), 2, # shared with maternal stool only
                                         -1))) 
}

overlap <- asv_infant
for(i in 1:ncol(overlap)){
  overlap[, i] <- ifelse((asv_vagina[,i]>0 & asv_stool[,i]>0), 4, -1)
}


## Definition 2: When present is defined as count >= 1count/1000sequence
sequence_infant <- select(infant_id, c("sample.id", "paired_ID", "m16s_slv138_97_depth"))
sequence_infant <- sequence_infant[order(sequence_infant$'sample.id'),]

sequence_vagina <- select(vagina_id, c("sample.id", "paired_ID", "m16s_slv138_97_depth"))
sequence_vagina <- sequence_vagina[order(sequence_vagina$'sample.id'),]

sequence_stool <- select(stool_id, c("sample.id", "paired_ID", "m16s_slv138_97_depth"))
sequence_stool <- sequence_stool[order(sequence_stool$'sample.id'),]

share_res_sd <- asv_infant

for(i in 1:ncol(share_res_sd)){
  share_res_sd[, i] <- ifelse(((asv_infant[,i]/sequence_infant[,3]>=0.001) & (asv_vagina[,i]/sequence_vagina[,3]>=0.001) & (asv_stool[,i]/sequence_stool[,3]>=0.001)), 3, # shared with both maternal vagina and stool
                           ifelse(((asv_infant[,i]/sequence_infant[,3]>=0.001) & (asv_vagina[,i]/sequence_vagina[,3]>=0.001) & (asv_stool[,i]/sequence_stool[,3]<0.001)), 1, # shared with maternal vagina only
                                  ifelse(((asv_infant[,i]/sequence_infant[,3]>=0.001) & (asv_vagina[,i]/sequence_vagina[,3]<0.001) & (asv_stool[,i]/sequence_stool[,3]>=0.001)), 2, # shared with maternal stool only
                                         -1))) 
}

overlap_sd <- asv_infant
for(i in 1:ncol(overlap_sd)){
  overlap_sd[, i] <- ifelse((asv_vagina[,i]/sequence_vagina[,3]>=0.001 & asv_stool[,i]/sequence_stool[,3]>=0.001), 4, -1)
}


## Definition 3: When present is defined as relative abundance >=0.05%
cut <- 0.05
asv_infant_rlt <- asv_infant / rowSums(asv_infant) * 100
asv_vagina_rlt <- asv_vagina / rowSums(asv_vagina) * 100
asv_stool_rlt <- asv_stool / rowSums(asv_stool) * 100

share_res_rlt_0.05 <- asv_infant
for(i in 1:ncol(share_res_rlt_0.05)){
  share_res_rlt_0.05[, i] <- ifelse(asv_infant_rlt[,i]>=cut & (asv_vagina_rlt[,i]>=cut) & (asv_stool_rlt[,i]>=cut), 3, # shared with both maternal vagina and stool
                           ifelse(asv_infant_rlt[,i]>=cut & (asv_vagina_rlt[,i]>=cut) & (asv_stool_rlt[,i]<cut), 1, # shared with maternal vagina only
                                  ifelse(asv_infant_rlt[,i]>=cut & (asv_vagina_rlt[,i]<cut) & (asv_stool_rlt[,i]>=cut), 2, # shared with maternal fecal only 
                                         -1)))  
}

overlap_rlt_0.05 <- asv_infant
for(i in 1:ncol(overlap_rlt_0.05)){
  overlap_rlt_0.05[, i] <- ifelse((asv_vagina_rlt[,i]>=cut & asv_stool_rlt[,i]>=cut), 4, -1)
}

## Definition 4: When present is defined as relative abundance >=0.01%
cut <- 0.01

share_res_rlt_0.01 <- asv_infant
for(i in 1:ncol(share_res_rlt_0.01)){
  share_res_rlt_0.01[, i] <- ifelse(asv_infant_rlt[,i]>=cut & (asv_vagina_rlt[,i]>=cut) & (asv_stool_rlt[,i]>=cut), 3, # shared with both maternal vagina and stool
                           ifelse(asv_infant_rlt[,i]>=cut & (asv_vagina_rlt[,i]>=cut) & (asv_stool_rlt[,i]<cut), 1, # shared with maternal vagina only
                                  ifelse(asv_infant_rlt[,i]>=cut & (asv_vagina_rlt[,i]<cut) & (asv_stool_rlt[,i]>=cut), 2, # shared with maternal fecal only 
                                         -1)))  
}

overlap_rlt_0.01 <- asv_infant
for(i in 1:ncol(overlap_rlt_0.01)){
  overlap_rlt_0.01[, i] <- ifelse((asv_vagina_rlt[,i]>=cut & asv_stool_rlt[,i]>=cut), 4, -1)
}

```

# Examine shared ASVs by infant
## Create a dataset to summarize results
```{r}
## When present is defined as count>=1
share_res_byinfant <- as.data.frame(infant_id[, c("sample.id", "ParticipantID", "paired_ID")])
share_res_byinfant <- share_res_byinfant[order(share_res_byinfant$sample.id), ]

share_res_byinfant <- share_res_byinfant %>%
  mutate(
    # count how many taxa are shared 
    both = rowSums(share_res==3), 
    vagina = rowSums(share_res==1),
    stool = rowSums(share_res==2),
    overlap = rowSums(overlap==4), 
    
    # count how many taxa existed in mother and infant
    sum_infant = rowSums(asv_infant>0),
    sum_vagina = rowSums(asv_vagina>0),
    sum_stool = rowSums(asv_stool>0)
  ) 

## When present is defined as count >= 1count/1000sequence
share_res_byinfant_sd <- as.data.frame(infant_id[, c("sample.id", "ParticipantID", "paired_ID")]) 
share_res_byinfant_sd <- share_res_byinfant_sd[order(share_res_byinfant_sd$sample.id), ]

share_res_byinfant_sd <- share_res_byinfant_sd %>%
  mutate(
    # count how many taxa are shared 
    both = rowSums(share_res_sd==3), 
    vagina = rowSums(share_res_sd==1),
    stool = rowSums(share_res_sd==2),
    overlap = rowSums(overlap_sd==4), 
    
    # count how many taxa existed in mother and infant
    sum_infant = rowSums(asv_infant/sequence_infant[,3]>=0.001),
    sum_vagina = rowSums(asv_vagina/sequence_vagina[,3]>=0.001),
    sum_stool = rowSums(asv_stool/sequence_stool[,3]>=0.001)
  ) 

## When present is defined as relative abundance >=0.05%
share_res_byinfant_rlt_0.05 <- as.data.frame(infant_id[, c("sample.id", "ParticipantID", "paired_ID")]) 
share_res_byinfant_rlt_0.05 <- share_res_byinfant_rlt_0.05[order(share_res_byinfant_rlt_0.05$sample.id), ]

share_res_byinfant_rlt_0.05 <- share_res_byinfant_rlt_0.05 %>%
  mutate(
    # count how many taxa are shared 
    both = rowSums(share_res_rlt_0.05==3), 
    vagina = rowSums(share_res_rlt_0.05==1),
    stool = rowSums(share_res_rlt_0.05==2),
    overlap = rowSums(overlap_rlt_0.05==4), 
    
    # count how many taxa existed in mother and infant
    sum_infant = rowSums(asv_infant_rlt>=0.05),
    sum_vagina = rowSums(asv_vagina_rlt>=0.05),
    sum_stool = rowSums(asv_stool_rlt>=0.05)
  ) 


## When present is defined as relative abundance >=0.01%
share_res_byinfant_rlt_0.01 <- as.data.frame(infant_id[, c("sample.id", "ParticipantID", "paired_ID")]) 
share_res_byinfant_rlt_0.01 <- share_res_byinfant_rlt_0.01[order(share_res_byinfant_rlt_0.01$sample.id), ]

share_res_byinfant_rlt_0.01 <- share_res_byinfant_rlt_0.01 %>%
  mutate(
    # count how many taxa are shared 
    both = rowSums(share_res_rlt_0.01==3), 
    vagina = rowSums(share_res_rlt_0.01==1),
    stool = rowSums(share_res_rlt_0.01==2),
    overlap = rowSums(overlap_rlt_0.01==4), 
    
    # count how many taxa existed in mother and infant
    sum_infant = rowSums(asv_infant_rlt>=0.01),
    sum_vagina = rowSums(asv_vagina_rlt>=0.01),
    sum_stool = rowSums(asv_stool_rlt>=0.01)
  ) 



# Add covariates
trimester1 <- filter(march, body_site=="vaginal")  # maternal vaginal samples
trimester1 <- select(trimester1, c(paired_ID, m16s_coll_tmst))
colnames(trimester1)[2] <- "m16s_coll_tmst" 

trimester2 <- filter(march, participant_type=="biological mother" & body_site=="stool")  # maternal fecal samples
trimester2 <- select(trimester2, c(paired_ID, m16s_coll_tmst))
colnames(trimester2)[2] <- "m16s_coll_tmst_f" 

trimester <- merge(trimester1, trimester2, by=c("paired_ID"))


share_res_byinfant <- merge(share_res_byinfant, master_dt[, -c(1)], by.x="ParticipantID", by.y="ParticipantID.x", all=FALSE)
share_res_byinfant_sd <- merge(share_res_byinfant_sd, master_dt[, -c(1)], by.x="ParticipantID", by.y="ParticipantID.x", all=FALSE)
share_res_byinfant_rlt_0.05 <- merge(share_res_byinfant_rlt_0.05, master_dt[, -c(1)], by.x="ParticipantID", by.y="ParticipantID.x", all=FALSE)
share_res_byinfant_rlt_0.01 <- merge(share_res_byinfant_rlt_0.01, master_dt[, -c(1)], by.x="ParticipantID", by.y="ParticipantID.x", all=FALSE)

share_res_byinfant <- merge(share_res_byinfant, trimester, by= c("paired_ID"))
share_res_byinfant$ParticipantID.y <- NULL

share_res_byinfant_sd <- merge(share_res_byinfant_sd, trimester, by= c("paired_ID"))
share_res_byinfant_sd$ParticipantID.y <- NULL

share_res_byinfant_rlt_0.05 <- merge(share_res_byinfant_rlt_0.05, trimester, by= c("paired_ID"))
share_res_byinfant_rlt_0.05$ParticipantID.y <- NULL

share_res_byinfant_rlt_0.01 <- merge(share_res_byinfant_rlt_0.01, trimester, by= c("paired_ID"))
share_res_byinfant_rlt_0.01$ParticipantID.y <- NULL


share_res_byinfant <- merge(share_res_byinfant, march[, c("sample.id", "feeding", "bf")], by=c("sample.id"))
share_res_byinfant_sd <- merge(share_res_byinfant_sd, march[, c("sample.id", "feeding", "bf")], by=c("sample.id"))
share_res_byinfant_rlt_0.05 <- merge(share_res_byinfant_rlt_0.05, march[, c("sample.id", "feeding", "bf")], by=c("sample.id"))
share_res_byinfant_rlt_0.01 <- merge(share_res_byinfant_rlt_0.01, march[, c("sample.id", "feeding", "bf")], by=c("sample.id"))


# extract the id for the 1 infant with missing delivery mode
share_res_byinfant[is.na(share_res_byinfant$birth_delivm), ]

# exclude 1 infant with missing delivery mode
share_res_byinfant <- filter(share_res_byinfant, !is.na(birth_delivm))
share_res_byinfant_sd <- filter(share_res_byinfant_sd, !is.na(birth_delivm))
share_res_byinfant_rlt_0.05 <- filter(share_res_byinfant_rlt_0.05, !is.na(birth_delivm))
share_res_byinfant_rlt_0.01 <- filter(share_res_byinfant_rlt_0.01, !is.na(birth_delivm))

```


## Modify dataset for visualization and statistical tests
```{r}
# store a wide dataset for later use when run Wilcoxon tests
share_res_byinfant_wide <- share_res_byinfant
share_res_byinfant_wide_sd <- share_res_byinfant_sd
share_res_byinfant_wide_rlt_0.05 <- share_res_byinfant_rlt_0.05
share_res_byinfant_wide_rlt_0.01 <- share_res_byinfant_rlt_0.01
```

## Visualize ASV overlapping
### Definition 1: When present is defined as count>=1

```{r fig.width=12, fig.height=12}
p_overall <- venn_3group(share_res_byinfant_wide)
p_1st <- venn_3group(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==1, ])
p_2nd <- venn_3group(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==2, ])
p_3rd <- venn_3group(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==3, ])

ggarrange(p_overall, p_1st, p_2nd, p_3rd, nrow=2, ncol=2, labels = c("Overall", "1st", "2nd", "3rd"))
```


### Definition 2: When present is defined as count>=1count/1000sequences
```{r fig.width=12, fig.height=12}
p_overall <- venn_3group(share_res_byinfant_wide_sd)
p_1st <- venn_3group(share_res_byinfant_wide_sd[share_res_byinfant_wide_sd$m16s_coll_tmst==1, ])
p_2nd <- venn_3group(share_res_byinfant_wide_sd[share_res_byinfant_wide_sd$m16s_coll_tmst==2, ])
p_3rd <- venn_3group(share_res_byinfant_wide_sd[share_res_byinfant_wide_sd$m16s_coll_tmst==3, ])

ggarrange(p_overall, p_1st, p_2nd, p_3rd, nrow=2, ncol=2, labels = c("Overall", "1st", "2nd", "3rd"))
```

### Definition 3: When present is defined as relative abundance >=0.05%
```{r fig.width=12, fig.height=12}
p_overall <- venn_3group(share_res_byinfant_wide_rlt_0.05)
p_1st <- venn_3group(share_res_byinfant_wide_rlt_0.05[share_res_byinfant_wide_rlt_0.05$m16s_coll_tmst==1, ])
p_2nd <- venn_3group(share_res_byinfant_wide_rlt_0.05[share_res_byinfant_wide_rlt_0.05$m16s_coll_tmst==2, ])
p_3rd <- venn_3group(share_res_byinfant_wide_rlt_0.05[share_res_byinfant_wide_rlt_0.05$m16s_coll_tmst==3, ])

ggarrange(p_overall, p_1st, p_2nd, p_3rd, nrow=2, ncol=2, labels = c("Overall", "1st", "2nd", "3rd"))
```

### Definition 4: When present is defined as relative abundance >=0.01%
```{r fig.width=12, fig.height=12}
p_overall <- venn_3group(share_res_byinfant_wide_rlt_0.01)
p_1st <- venn_3group(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==1, ])
p_2nd <- venn_3group(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==2, ])
p_3rd <- venn_3group(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==3, ])

ggarrange(p_overall, p_1st, p_2nd, p_3rd, nrow=2, ncol=2, labels = c("Overall", "1st", "2nd", "3rd"))
```
   
#### I will use this definition to account for sequencing depth moving forward. 
   
  
## Test on absolute count of shared ASVs using paired Wilcox test, overall and by trimester
### Use raw ASV data
```{r}
share_res_byinfant_wide <- share_res_byinfant_wide %>%
  mutate(sum_shared = stool + vagina + both, 
         sum_not_shared = sum_infant - stool - vagina - both)

data <- share_res_byinfant_wide

## does number of shared and not shared ASV differ, overall and by trimester?
wilcox.test(data$sum_not_shared, data$sum_shared, paired = TRUE)
wilcox.test(data[data$m16s_coll_tmst==1,]$sum_not_shared, data[data$m16s_coll_tmst==1,]$sum_shared, paired = TRUE)
wilcox.test(data[data$m16s_coll_tmst==2,]$sum_not_shared, data[data$m16s_coll_tmst==2,]$sum_shared, paired = TRUE)
wilcox.test(data[data$m16s_coll_tmst==3,]$sum_not_shared, data[data$m16s_coll_tmst==3,]$sum_shared, paired = TRUE)


## does number of shared ASV with maternal stool and vagina differ, overall and by trimester?
wilcox.test(data$stool, data$vagina, paired = TRUE)
wilcox.test(data[data$m16s_coll_tmst==1,]$stool, data[data$m16s_coll_tmst==1,]$vagina, paired = TRUE)
wilcox.test(data[data$m16s_coll_tmst==2,]$stool, data[data$m16s_coll_tmst==2,]$vagina, paired = TRUE)
wilcox.test(data[data$m16s_coll_tmst==3,]$stool, data[data$m16s_coll_tmst==3,]$vagina, paired = TRUE)
```


### Use adjusted ASV data
```{r}
share_res_byinfant_wide_rlt_0.01 <- share_res_byinfant_wide_rlt_0.01 %>%
  mutate(sum_shared = stool + vagina + both, 
         sum_not_shared = sum_infant - stool - vagina - both)

data <- share_res_byinfant_wide_rlt_0.01

## does number of shared and not shared ASV differ, overall and by trimester?
wilcox.test(data$sum_not_shared, data$sum_shared, paired = TRUE)
wilcox.test(data[data$m16s_coll_tmst==1,]$sum_not_shared, data[data$m16s_coll_tmst==1,]$sum_shared, paired = TRUE)
wilcox.test(data[data$m16s_coll_tmst==2,]$sum_not_shared, data[data$m16s_coll_tmst==2,]$sum_shared, paired = TRUE)
wilcox.test(data[data$m16s_coll_tmst==3,]$sum_not_shared, data[data$m16s_coll_tmst==3,]$sum_shared, paired = TRUE)


## does number of shared ASV with maternal stool and vagina differ, overall and by trimester?
wilcox.test(data$stool, data$vagina, paired = TRUE)
wilcox.test(data[data$m16s_coll_tmst==1,]$stool, data[data$m16s_coll_tmst==1,]$vagina, paired = TRUE)
wilcox.test(data[data$m16s_coll_tmst==2,]$stool, data[data$m16s_coll_tmst==2,]$vagina, paired = TRUE)
wilcox.test(data[data$m16s_coll_tmst==3,]$stool, data[data$m16s_coll_tmst==3,]$vagina, paired = TRUE)
```



## Focus on 3rd trimester data - by delivery mode, antibiotics, feeding
```{r}
table(share_res_byinfant_wide$m16s_coll_tmst, share_res_byinfant_wide$birth_delivm)
table(share_res_byinfant_wide$m16s_coll_tmst, share_res_byinfant_wide$med_antib_pn)
table(share_res_byinfant_wide$m16s_coll_tmst, share_res_byinfant_wide$bf)
```

### Use raw ASV data
```{r fig.height=8, fig.width=15, message=FALSE}
# convert data from wide to long
share_res_byinfant <- gather(share_res_byinfant_wide, type, n_count, c(both:stool, sum_not_shared), factor_key=TRUE)

share_res_byinfant$type <- factor(share_res_byinfant$type, 
                                  levels=c("sum_not_shared","stool","both", "vagina"),
                                  labels=c("Not shared with either maternal vaginal or fecal microbiota", "Shared with maternal fecal microbiota only", "Shared with both maternal fecal and vaginal microbiota", "Shared with maternal vaginal microbiota only"))

share_res_byinfant$birth_delivm <- factor(share_res_byinfant$birth_delivm, levels = c("Vaginal delivery", "Cesarean delivery"), 
                                          labels=c("Vaginal", "C-section"))
p_deliv <- 
  share_res_byinfant[share_res_byinfant$m16s_coll_tmst==3,] %>%
  group_by(birth_delivm, type) %>%
  summarise(mean = mean(n_count), prop = mean(n_count)/mean(sum_infant)*100) %>%
  ggplot(aes(x = birth_delivm, y = prop, fill = type)) +
    geom_col(aes(linetype=birth_delivm), color="black", lwd=1) + 
    scale_x_discrete(limits = c(" ","C-section", "Vaginal")) +
    coord_polar(theta = "y") + labs(fill="Sharing", linetype="Delivery Mode") +
    scale_fill_manual(values=c("white","#0072B2","#90C987","#E69F00")) +
    geom_text(aes(label = paste0(round(mean, 0), "\n(", round(prop, 1), "%)")),
              position = position_stack(vjust = 0.5), size=5) +
    theme_void() + theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin(), 
                         legend.title = element_text(face="bold", size=17), legend.text = element_text(size=17)) + guides(fill = FALSE) 

share_res_byinfant$med_antib_pn <- factor(share_res_byinfant$med_antib_pn, levels=c("No antibiotics", "Took antibiotics"),
                                          labels=c("No", "Yes"))
p_antib <- 
  share_res_byinfant[share_res_byinfant$m16s_coll_tmst==3,] %>%
  group_by(med_antib_pn, type) %>%
  summarise(mean = mean(n_count), prop = mean(n_count)/mean(sum_infant)*100) %>%
  ggplot(aes(x = med_antib_pn, y = prop, fill = type)) +
    geom_col(aes(linetype=med_antib_pn), color="black", lwd=1) + 
    scale_x_discrete(limits = c(" ","Yes", "No")) +
    coord_polar(theta = "y") + labs(fill="Sharing", linetype="Prenatal Antibiotic Use") +
    scale_fill_manual(values=c("white","#0072B2","#90C987","#E69F00")) +
    geom_text(aes(label = paste0(round(mean, 0), "\n(", round(prop, 1), "%)")),
              position = position_stack(vjust = 0.5), size=5) +
    theme_void() + theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin(), 
                         legend.title = element_text(face="bold", size=17), legend.text = element_text(size=17)) + guides(fill=FALSE)

p_bf <- 
  share_res_byinfant[share_res_byinfant$m16s_coll_tmst==3,] %>%
  group_by(bf, type) %>%
  summarise(mean = mean(n_count), prop = mean(n_count)/mean(sum_infant)*100) %>%
  ggplot(aes(x = bf, y = prop, fill = type)) +
    geom_col(aes(linetype=bf), color="black", lwd=1) + 
    scale_x_discrete(limits = c(" ","No", "Yes")) +
    coord_polar(theta = "y") + labs(fill="Sharing", linetype="Ever Breastfed") +
    scale_fill_manual(values=c("white","#0072B2","#90C987","#E69F00")) +
    geom_text(aes(label = paste0(round(mean, 0), "\n(", round(prop, 1), "%)")),
              position = position_stack(vjust = 0.5), size=5) +
    theme_void() + theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin(), 
                         legend.title = element_text(face="bold", size=17), legend.text = element_text(size=17)) + guides(fill=FALSE)


p <- ggarrange(p_antib, p_deliv, p_bf, nrow=1) 
p
```
```{r eval=FALSE, include=FALSE}
ggsave(p, filename = "MARCH_3rdtms_ASV_byvariables_raw_09262323.png",
       path =  paste0(path, "Figures"),
       height = 8, width = 15, units = "in", dpi = 300)
```

#### Test with each group
```{r}
data <- filter(share_res_byinfant_wide, m16s_coll_tmst==3)
fun_wilcoxon_paired_MARCH_2(data)
```

#### Test across groups
```{r message=FALSE, warning=FALSE, fig.height=6.5, fig.width=20}
wide_data <- filter(share_res_byinfant_wide, m16s_coll_tmst==3)

p <- test_ASV_byvariabels(wide_data)
p
```
```{r eval=FALSE, include=FALSE}
ggsave(p, filename = "MARCH_3rdtms_ASV_betweenvariables_raw_09262323.png",
       path =  paste0(path, "Figures"),
       height = 6.5, width = 20, units = "in", dpi = 300)

```

```{r fig.height=8, fig.width=10, eval=FALSE, include=FALSE}
p_bf_2 <-
  share_res_byinfant[share_res_byinfant$m16s_coll_tmst==3,] %>%
  group_by(bf, type) %>%
  summarise(mean = mean(n_count), prop = mean(n_count)/mean(sum_infant)*100) %>%
  ggplot(aes(x = bf, y = prop, fill = type)) +
    geom_col(aes(linetype=bf), color="black", lwd=1) + 
    scale_x_discrete(limits = c(" ","No", "Yes")) +
    coord_polar(theta = "y") + labs(fill="Sharing", linetype="Ever Breastfed") +
    scale_fill_manual(values=c("white","#0072B2","#90C987","#E69F00")) +
    geom_text(aes(label = paste0(round(mean, 0), "\n(", round(prop, 1), "%)")),
              position = position_stack(vjust = 0.5), size=5) +
    theme_void() + theme(legend.position = "right", legend.direction = "vertical", legend.margin=margin(), 
                         legend.title = element_text(face="bold", size=17), legend.text = element_text(size=17)) +
    guides(linetype=FALSE)

ggsave(p_bf_2, filename = "MARCH_3rdtms_ASV_byvariables_legend_09262323.png",
       path =  paste0(path, "Figures"),
       height = 8, width = 10, units = "in", dpi = 300)

```

### Use adjusted ASV data
```{r fig.height=8, fig.width=15, message=FALSE}
# convert data from wide to long
share_res_byinfant_rlt_0.01 <- gather(share_res_byinfant_wide_rlt_0.01, type, n_count, c(both:stool, sum_not_shared), factor_key=TRUE)

share_res_byinfant_rlt_0.01$type <- factor(share_res_byinfant_rlt_0.01$type, 
                                  levels=c("sum_not_shared","stool","both", "vagina"),
                                  labels=c("Not shared with either maternal vaginal or fecal microbiota", "Shared with maternal fecal microbiota only", "Shared with both maternal fecal and vaginal microbiota", "Shared with maternal vaginal microbiota only"))

share_res_byinfant_rlt_0.01$birth_delivm <- factor(share_res_byinfant_rlt_0.01$birth_delivm, 
                                                   levels = c("Vaginal delivery", "Cesarean delivery"), 
                                          labels=c("Vaginal", "C-section"))
p_deliv <- 
  share_res_byinfant_rlt_0.01[share_res_byinfant_rlt_0.01$m16s_coll_tmst==3,] %>%
  group_by(birth_delivm, type) %>%
  summarise(mean = mean(n_count), prop = mean(n_count)/mean(sum_infant)*100) %>%
  ggplot(aes(x = birth_delivm, y = prop, fill = type)) +
    geom_col(aes(linetype=birth_delivm), color="black", lwd=1) + 
    scale_x_discrete(limits = c(" ","C-section", "Vaginal")) +
    coord_polar(theta = "y") + labs(fill="Sharing", linetype="Delivery Mode") +
    scale_fill_manual(values=c("white","#0072B2","#90C987","#E69F00")) +
    geom_text(aes(label = paste0(round(mean, 0), "\n(", round(prop, 1), "%)")),
              position = position_stack(vjust = 0.5), size=5) +
    theme_void() + theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin(), 
                         legend.title = element_text(face="bold", size=17), legend.text = element_text(size=17)) + guides(fill = FALSE) 

share_res_byinfant_rlt_0.01$med_antib_pn <- factor(share_res_byinfant_rlt_0.01$med_antib_pn, 
                                                   levels=c("No antibiotics", "Took antibiotics"),
                                          labels=c("No", "Yes"))
p_antib <- 
  share_res_byinfant_rlt_0.01[share_res_byinfant_rlt_0.01$m16s_coll_tmst==3,] %>%
  group_by(med_antib_pn, type) %>%
  summarise(mean = mean(n_count), prop = mean(n_count)/mean(sum_infant)*100) %>%
  ggplot(aes(x = med_antib_pn, y = prop, fill = type)) +
    geom_col(aes(linetype=med_antib_pn), color="black", lwd=1) + 
    scale_x_discrete(limits = c(" ","Yes", "No")) +
    coord_polar(theta = "y") + labs(fill="Sharing", linetype="Prenatal Antibiotic Use") +
    scale_fill_manual(values=c("white","#0072B2","#90C987","#E69F00")) +
    geom_text(aes(label = paste0(round(mean, 0), "\n(", round(prop, 1), "%)")),
              position = position_stack(vjust = 0.5), size=5) +
    theme_void() + theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin(), 
                         legend.title = element_text(face="bold", size=17), legend.text = element_text(size=17)) + guides(fill=FALSE)

p_bf <- 
  share_res_byinfant_rlt_0.01[share_res_byinfant_rlt_0.01$m16s_coll_tmst==3,] %>%
  group_by(bf, type) %>%
  summarise(mean = mean(n_count), prop = mean(n_count)/mean(sum_infant)*100) %>%
  ggplot(aes(x = bf, y = prop, fill = type)) +
    geom_col(aes(linetype=bf), color="black", lwd=1) + 
    scale_x_discrete(limits = c(" ","No", "Yes")) +
    coord_polar(theta = "y") + labs(fill="Sharing", linetype="Ever Breastfed") +
    scale_fill_manual(values=c("white","#0072B2","#90C987","#E69F00")) +
    geom_text(aes(label = paste0(round(mean, 0), "\n(", round(prop, 1), "%)")),
              position = position_stack(vjust = 0.5), size=5) +
    theme_void() + theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin(), 
                         legend.title = element_text(face="bold", size=17), legend.text = element_text(size=17)) + guides(fill=FALSE)


p_adj <- ggarrange(p_antib, p_deliv, p_bf, nrow=1) 
p_adj
```
```{r eval=FALSE, include=FALSE}
ggsave(p_adj, filename = "MARCH_3rdtms_ASV_byvariables_adj_09262323.png",
       path =  paste0(path, "Figures"),
       height = 8, width = 15, units = "in", dpi = 300)
```

#### Test
```{r warning=FALSE}
data <- filter(share_res_byinfant_wide_rlt_0.01, m16s_coll_tmst==3)
fun_wilcoxon_paired_MARCH_2(data)
```

#### Test across groups
```{r message=FALSE, warning=FALSE, fig.height=6.5, fig.width=20}
wide_data <- filter(share_res_byinfant_wide_rlt_0.01, m16s_coll_tmst==3)

p <- test_ASV_byvariabels(wide_data)
p
```
```{r eval=FALSE, include=FALSE}
ggsave(p, filename = "MARCH_3rdtms_ASV_betweenvariables_adj_09262323.png",
       path =  paste0(path, "Figures"),
       height = 6.5, width = 20, units = "in", dpi = 300)

```

## Focus on same trimester data - by delivery mode, antibiotics, feeding
```{r}
table(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==share_res_byinfant_wide$m16s_coll_tmst_f,]$birth_delivm)
table(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==share_res_byinfant_wide$m16s_coll_tmst_f,]$med_antib_pn)
table(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==share_res_byinfant_wide$m16s_coll_tmst_f,]$bf)
```

### Use raw ASV data
```{r fig.height=8, fig.width=15, message=FALSE}
p_deliv <- 
  share_res_byinfant[share_res_byinfant$m16s_coll_tmst==share_res_byinfant$m16s_coll_tmst_f,] %>%
  group_by(birth_delivm, type) %>%
  summarise(mean = mean(n_count), prop = mean(n_count)/mean(sum_infant)*100) %>%
  ggplot(aes(x = birth_delivm, y = prop, fill = type)) +
    geom_col(aes(linetype=birth_delivm), color="black", lwd=1) + 
    scale_x_discrete(limits = c(" ","C-section", "Vaginal")) +
    coord_polar(theta = "y") + labs(fill="Sharing", linetype="Delivery Mode") +
    scale_fill_manual(values=c("white","#0072B2","#90C987","#E69F00")) +
    geom_text(aes(label = paste0(round(mean, 0), "\n(", round(prop, 1), "%)")),
              position = position_stack(vjust = 0.5), size=5) +
    theme_void() + theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin(), 
                         legend.title = element_text(face="bold", size=17), legend.text = element_text(size=17)) + guides(fill = FALSE) 

p_antib <- 
  share_res_byinfant[share_res_byinfant$m16s_coll_tmst==share_res_byinfant$m16s_coll_tmst_f,] %>%
  group_by(med_antib_pn, type) %>%
  summarise(mean = mean(n_count), prop = mean(n_count)/mean(sum_infant)*100) %>%
  ggplot(aes(x = med_antib_pn, y = prop, fill = type)) +
    geom_col(aes(linetype=med_antib_pn), color="black", lwd=1) + 
    scale_x_discrete(limits = c(" ","Yes", "No")) +
    coord_polar(theta = "y") + labs(fill="Sharing", linetype="Prenatal Antibiotic Use") +
    scale_fill_manual(values=c("white","#0072B2","#90C987","#E69F00")) +
    geom_text(aes(label = paste0(round(mean, 0), "\n(", round(prop, 1), "%)")),
              position = position_stack(vjust = 0.5), size=5) +
    theme_void() + theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin(), 
                         legend.title = element_text(face="bold", size=17), legend.text = element_text(size=17)) + guides(fill=FALSE)

p_bf <- 
  share_res_byinfant[share_res_byinfant$m16s_coll_tmst==share_res_byinfant$m16s_coll_tmst_f,] %>%
  group_by(bf, type) %>%
  summarise(mean = mean(n_count), prop = mean(n_count)/mean(sum_infant)*100) %>%
  ggplot(aes(x = bf, y = prop, fill = type)) +
    geom_col(aes(linetype=bf), color="black", lwd=1) + 
    scale_x_discrete(limits = c(" ","No", "Yes")) +
    coord_polar(theta = "y") + labs(fill="Sharing", linetype="Ever Breastfed") +
    scale_fill_manual(values=c("white","#0072B2","#90C987","#E69F00")) +
    geom_text(aes(label = paste0(round(mean, 0), "\n(", round(prop, 1), "%)")),
              position = position_stack(vjust = 0.5), size=5) +
    theme_void() + theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin(), 
                         legend.title = element_text(face="bold", size=17), legend.text = element_text(size=17)) + guides(fill=FALSE)


p <- ggarrange(p_antib, p_deliv, p_bf, nrow=1) 
p
```
```{r eval=FALSE, include=FALSE}
ggsave(p, filename = "MARCH_sametms_ASV_byvariables_raw_09262323.png",
       path =  paste0(path, "Figures"),
       height = 8, width = 15, units = "in", dpi = 300)
```

#### Test
```{r warning=FALSE}
data <- filter(share_res_byinfant_wide, m16s_coll_tmst==m16s_coll_tmst_f)
fun_wilcoxon_paired_MARCH_2(data)
```
   
#### Test across groups
```{r message=FALSE, warning=FALSE, fig.height=6.5, fig.width=20}
wide_data <- filter(share_res_byinfant_wide, m16s_coll_tmst==m16s_coll_tmst_f)

p <- test_ASV_byvariabels(wide_data)
p
```
```{r eval=FALSE, include=FALSE}
ggsave(p, filename = "MARCH_sametms_ASV_betweenvariables_raw_09262323.png",
       path =  paste0(path, "Figures"),
       height = 6.5, width = 20, units = "in", dpi = 300)

```

### Use adjusted ASV data
```{r fig.height=8, fig.width=15, message=FALSE}
p_deliv <- 
  share_res_byinfant_rlt_0.01[share_res_byinfant_rlt_0.01$m16s_coll_tmst==share_res_byinfant_rlt_0.01$m16s_coll_tmst_f,] %>%
  group_by(birth_delivm, type) %>%
  summarise(mean = mean(n_count), prop = mean(n_count)/mean(sum_infant)*100) %>%
  ggplot(aes(x = birth_delivm, y = prop, fill = type)) +
    geom_col(aes(linetype=birth_delivm), color="black", lwd=1) + 
    scale_x_discrete(limits = c(" ","C-section", "Vaginal")) +
    coord_polar(theta = "y") + labs(fill="Sharing", linetype="Delivery Mode") +
    scale_fill_manual(values=c("white","#0072B2","#90C987","#E69F00")) +
    geom_text(aes(label = paste0(round(mean, 0), "\n(", round(prop, 1), "%)")),
              position = position_stack(vjust = 0.5), size=5) +
    theme_void() + theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin(), 
                         legend.title = element_text(face="bold", size=17), legend.text = element_text(size=17)) + guides(fill = FALSE) 

p_antib <- 
  share_res_byinfant_rlt_0.01[share_res_byinfant_rlt_0.01$m16s_coll_tmst==share_res_byinfant_rlt_0.01$m16s_coll_tmst_f,] %>%
  group_by(med_antib_pn, type) %>%
  summarise(mean = mean(n_count), prop = mean(n_count)/mean(sum_infant)*100) %>%
  ggplot(aes(x = med_antib_pn, y = prop, fill = type)) +
    geom_col(aes(linetype=med_antib_pn), color="black", lwd=1) + 
    scale_x_discrete(limits = c(" ","Yes", "No")) +
    coord_polar(theta = "y") + labs(fill="Sharing", linetype="Prenatal Antibiotic Use") +
    scale_fill_manual(values=c("white","#0072B2","#90C987","#E69F00")) +
    geom_text(aes(label = paste0(round(mean, 0), "\n(", round(prop, 1), "%)")),
              position = position_stack(vjust = 0.5), size=5) +
    theme_void() + theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin(), 
                         legend.title = element_text(face="bold", size=17), legend.text = element_text(size=17)) + guides(fill=FALSE)

p_bf <- 
  share_res_byinfant_rlt_0.01[share_res_byinfant_rlt_0.01$m16s_coll_tmst==share_res_byinfant_rlt_0.01$m16s_coll_tmst_f,] %>%
  group_by(bf, type) %>%
  summarise(mean = mean(n_count), prop = mean(n_count)/mean(sum_infant)*100) %>%
  ggplot(aes(x = bf, y = prop, fill = type)) +
    geom_col(aes(linetype=bf), color="black", lwd=1) + 
    scale_x_discrete(limits = c(" ","No", "Yes")) +
    coord_polar(theta = "y") + labs(fill="Sharing", linetype="Ever Breastfed") +
    scale_fill_manual(values=c("white","#0072B2","#90C987","#E69F00")) +
    geom_text(aes(label = paste0(round(mean, 0), "\n(", round(prop, 1), "%)")),
              position = position_stack(vjust = 0.5), size=5) +
    theme_void() + theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin(), 
                         legend.title = element_text(face="bold", size=17), legend.text = element_text(size=17)) + guides(fill=FALSE)


p_adj <- ggarrange(p_antib, p_deliv, p_bf, nrow=1) 
p_adj
```
```{r eval=FALSE, include=FALSE}
ggsave(p_adj, filename = "MARCH_sametms_ASV_byvariables_adj_09262323.png",
       path =  paste0(path, "Figures"),
       height = 8, width = 15, units = "in", dpi = 300)
```

#### Test
```{r warning=FALSE}
data <- filter(share_res_byinfant_wide_rlt_0.01, m16s_coll_tmst==m16s_coll_tmst_f)
fun_wilcoxon_paired_MARCH_2(data)
```

#### Test across groups
```{r message=FALSE, warning=FALSE, fig.height=6.5, fig.width=20}
wide_data <- filter(share_res_byinfant_wide_rlt_0.01, m16s_coll_tmst==m16s_coll_tmst_f)

p <- test_ASV_byvariabels(wide_data)
p
```
```{r eval=FALSE, include=FALSE}
ggsave(p, filename = "MARCH_sametms_ASV_betweenvariables_adj_09262323.png",
       path =  paste0(path, "Figures"),
       height = 6.5, width = 20, units = "in", dpi = 300)

```


# Summarize shared ASVs by ASV
```{r}
# Need to summarize sharing seperately for each maternal source

## Definition 1: When present is defined as count>=1
asv_infant <- asv_infant[order(row.names(asv_infant)), ]
asv_vagina <- asv_vagina[order(row.names(asv_vagina)), ]
asv_stool <- asv_stool[order(row.names(asv_stool)), ]

share_res_asv_v <- asv_infant
for(i in 1:ncol(share_res_asv_v)){
  share_res_asv_v[, i] <- ifelse((asv_infant[,i]>0) & (asv_vagina[,i]>0), 1, # shared with maternal vagina 
                                 ifelse((asv_infant[,i]==0) & (asv_vagina[,i]==0), 0, # means not exist
                                                4)) # not shared (exist in either mother or infant)
}

share_res_asv_f <- asv_infant
for(i in 1:ncol(share_res_asv_f)){
  share_res_asv_f[, i] <- ifelse((asv_infant[,i]>0) & (asv_stool[,i]>0), 1, # shared with maternal fecal 
                                 ifelse((asv_infant[,i]==0) & (asv_stool[,i]==0), 0, # means not exist
                                                4)) # not shared (exist in either mother or infant)

}


## Definition 4: When present is defined as relative abundance >=0.01%
asv_infant_rlt <- asv_infant_rlt[order(row.names(asv_infant_rlt)), ]
asv_vagina_rlt <- asv_vagina_rlt[order(row.names(asv_vagina_rlt)), ]
asv_stool_rlt <- asv_stool_rlt[order(row.names(asv_stool_rlt)), ]

cut <- 0.01

share_res_asv_rlt_0.01_v <- asv_infant_rlt
for(i in 1:ncol(share_res_asv_rlt_0.01_v)){
  share_res_asv_rlt_0.01_v[, i] <- ifelse(asv_infant_rlt[,i]>=cut & (asv_vagina_rlt[,i]>=cut), 1, # shared with maternal vagina
                                          ifelse(asv_infant_rlt[,i]<cut & (asv_vagina_rlt[,i]<cut), 0, # means not exist
                                                4)) # not shared
}

share_res_asv_rlt_0.01_f <- asv_infant_rlt
for(i in 1:ncol(share_res_asv_rlt_0.01_f)){
  share_res_asv_rlt_0.01_f[, i] <- ifelse(asv_infant_rlt[,i]>=cut & (asv_stool_rlt[,i]>=cut), 1, # shared with maternal vagina
                                          ifelse(asv_infant_rlt[,i]<cut & (asv_stool_rlt[,i]<cut), 0, # means not exist
                                                4)) # not shared
}

```

## Create summary datasets
### 3rd trimester only
```{r}
#### Use raw ASV data: 3rd trimester only
# overall
id.list <- as.list(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==3,]$sample.id)
share_res_byasv <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_v, share_res_asv_f)
share_res_byasv$variable <- "Overall"


# by antibiotics
id.list <- as.list(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==3 & 
                                             share_res_byinfant_wide$med_antib_pn=="No antibiotics",]$sample.id)
junk1 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_v, share_res_asv_f)
junk1$variable <- "No antibiotics"

id.list <- as.list(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==3 & 
                                             share_res_byinfant_wide$med_antib_pn=="Took antibiotics",]$sample.id)
junk2 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_v, share_res_asv_f)
junk2$variable <- "Took antibiotics"


# by delivery mode
id.list <- as.list(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==3 & 
                                             share_res_byinfant_wide$birth_delivm=="Vaginal delivery",]$sample.id)
junk3 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_v, share_res_asv_f)
junk3$variable <- "Vaginal"

id.list <- as.list(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==3 & 
                                             share_res_byinfant_wide$birth_delivm=="Cesarean delivery",]$sample.id)
junk4 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_v, share_res_asv_f)
junk4$variable <- "C-section"


# by breastfeeding
id.list <- as.list(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==3 & 
                                             share_res_byinfant_wide$bf=="Yes",]$sample.id)
junk5 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_v, share_res_asv_f)
junk5$variable <- "Yes"

id.list <- as.list(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==3 & 
                                             share_res_byinfant_wide$bf=="No",]$sample.id)
junk6 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_v, share_res_asv_f)
junk6$variable <- "No"

# combine results
share_res_byasv <- rbind(share_res_byasv, junk1, junk2, junk3, junk4, junk5, junk6)
share_res_byasv$group <- factor(share_res_byasv$group, levels=c("Maternal fecal - Infant fecal", "Maternal vaginal - Infant fecal"),
                                labels=c("Sharing with maternal fecal microbiota", "Sharing with maternal vaginal microbiota"))
share_res_byasv$type <- factor(share_res_byasv$type, levels=c("Not shared", "Shared"))


#### Use adjusted ASV data: 3rd trimester only
# overall
id.list <- as.list(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==3,]$sample.id)
share_res_byasv_rlt_0.01 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_rlt_0.01_v, share_res_asv_rlt_0.01_f)
share_res_byasv_rlt_0.01$variable <- "Overall"


# by antibiotics
id.list <- as.list(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==3 & 
                                             share_res_byinfant_wide_rlt_0.01$med_antib_pn=="No antibiotics",]$sample.id)
junk1 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_rlt_0.01_v, share_res_asv_rlt_0.01_f)
junk1$variable <- "No antibiotics"

id.list <- as.list(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==3 & 
                                             share_res_byinfant_wide_rlt_0.01$med_antib_pn=="Took antibiotics",]$sample.id)
junk2 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_rlt_0.01_v, share_res_asv_rlt_0.01_f)
junk2$variable <- "Took antibiotics"


# by delivery mode
id.list <- as.list(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==3 & 
                                             share_res_byinfant_wide_rlt_0.01$birth_delivm=="Vaginal delivery",]$sample.id)
junk3 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_rlt_0.01_v, share_res_asv_rlt_0.01_f)
junk3$variable <- "Vaginal"

id.list <- as.list(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==3 & 
                                             share_res_byinfant_wide_rlt_0.01$birth_delivm=="Cesarean delivery",]$sample.id)
junk4 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_rlt_0.01_v, share_res_asv_rlt_0.01_f)
junk4$variable <- "C-section"


# by breastfeeding
id.list <- as.list(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==3 & 
                                             share_res_byinfant_wide_rlt_0.01$bf=="Yes",]$sample.id)
junk5 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_rlt_0.01_v, share_res_asv_rlt_0.01_f)
junk5$variable <- "Yes"

id.list <- as.list(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==3 & 
                                             share_res_byinfant_wide_rlt_0.01$bf=="No",]$sample.id)
junk6 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_rlt_0.01_v, share_res_asv_rlt_0.01_f)
junk6$variable <- "No"

# combine results
share_res_byasv_rlt_0.01 <- rbind(share_res_byasv_rlt_0.01, junk1, junk2, junk3, junk4, junk5, junk6)
share_res_byasv_rlt_0.01$group <- factor(share_res_byasv_rlt_0.01$group, 
                                         levels=c("Maternal fecal - Infant fecal", "Maternal vaginal - Infant fecal"),
                                labels=c("Sharing with maternal fecal microbiota", "Sharing with maternal vaginal microbiota"))
share_res_byasv_rlt_0.01$type <- factor(share_res_byasv_rlt_0.01$type, levels=c("Not shared", "Shared"))

```

### same trimester
```{r}
#### Use raw ASV data: same trimester only
# overall
id.list <- as.list(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==share_res_byinfant_wide$m16s_coll_tmst_f,]$sample.id)
share_res_byasv_2 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_v, share_res_asv_f)
share_res_byasv_2$variable <- "Overall"


# by antibiotics
id.list <- as.list(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==share_res_byinfant_wide$m16s_coll_tmst_f & 
                                             share_res_byinfant_wide$med_antib_pn=="No antibiotics",]$sample.id)
junk1 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_v, share_res_asv_f)
junk1$variable <- "No antibiotics"

id.list <- as.list(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==share_res_byinfant_wide$m16s_coll_tmst_f & 
                                             share_res_byinfant_wide$med_antib_pn=="Took antibiotics",]$sample.id)
junk2 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_v, share_res_asv_f)
junk2$variable <- "Took antibiotics"


# by delivery mode
id.list <- as.list(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==share_res_byinfant_wide$m16s_coll_tmst_f & 
                                             share_res_byinfant_wide$birth_delivm=="Vaginal delivery",]$sample.id)
junk3 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_v, share_res_asv_f)
junk3$variable <- "Vaginal"

id.list <- as.list(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==share_res_byinfant_wide$m16s_coll_tmst_f & 
                                             share_res_byinfant_wide$birth_delivm=="Cesarean delivery",]$sample.id)
junk4 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_v, share_res_asv_f)
junk4$variable <- "C-section"


# by breastfeeding
id.list <- as.list(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==share_res_byinfant_wide$m16s_coll_tmst_f & 
                                             share_res_byinfant_wide$bf=="Yes",]$sample.id)
junk5 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_v, share_res_asv_f)
junk5$variable <- "Yes"

id.list <- as.list(share_res_byinfant_wide[share_res_byinfant_wide$m16s_coll_tmst==share_res_byinfant_wide$m16s_coll_tmst_f & 
                                             share_res_byinfant_wide$bf=="No",]$sample.id)
junk6 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_v, share_res_asv_f)
junk6$variable <- "No"

# combine results
share_res_byasv_2 <- rbind(share_res_byasv_2, junk1, junk2, junk3, junk4, junk5, junk6)
share_res_byasv_2$group <- factor(share_res_byasv_2$group, levels=c("Maternal fecal - Infant fecal", "Maternal vaginal - Infant fecal"),
                                labels=c("Sharing with maternal fecal microbiota", "Sharing with maternal vaginal microbiota"))
share_res_byasv_2$type <- factor(share_res_byasv_2$type, levels=c("Not shared", "Shared"))


#### Use adjusted ASV data: same trimester only
# overall
id.list <- as.list(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst_f,]$sample.id)
share_res_byasv_rlt_0.01_2 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_rlt_0.01_v, share_res_asv_rlt_0.01_f)
share_res_byasv_rlt_0.01_2$variable <- "Overall"


# by antibiotics
id.list <- as.list(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst_f & 
                                             share_res_byinfant_wide_rlt_0.01$med_antib_pn=="No antibiotics",]$sample.id)
junk1 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_rlt_0.01_v, share_res_asv_rlt_0.01_f)
junk1$variable <- "No antibiotics"

id.list <- as.list(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst_f & 
                                             share_res_byinfant_wide_rlt_0.01$med_antib_pn=="Took antibiotics",]$sample.id)
junk2 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_rlt_0.01_v, share_res_asv_rlt_0.01_f)
junk2$variable <- "Took antibiotics"


# by delivery mode
id.list <- as.list(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst_f & 
                                             share_res_byinfant_wide_rlt_0.01$birth_delivm=="Vaginal delivery",]$sample.id)
junk3 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_rlt_0.01_v, share_res_asv_rlt_0.01_f)
junk3$variable <- "Vaginal"

id.list <- as.list(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst_f & 
                                             share_res_byinfant_wide_rlt_0.01$birth_delivm=="Cesarean delivery",]$sample.id)
junk4 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_rlt_0.01_v, share_res_asv_rlt_0.01_f)
junk4$variable <- "C-section"


# by breastfeeding
id.list <- as.list(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst_f & 
                                             share_res_byinfant_wide_rlt_0.01$bf=="Yes",]$sample.id)
junk5 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_rlt_0.01_v, share_res_asv_rlt_0.01_f)
junk5$variable <- "Yes"

id.list <- as.list(share_res_byinfant_wide_rlt_0.01[share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst==share_res_byinfant_wide_rlt_0.01$m16s_coll_tmst_f & 
                                             share_res_byinfant_wide_rlt_0.01$bf=="No",]$sample.id)
junk6 <- fun_sum_sharing_by_asv(asv_name, id.list, share_res_asv_rlt_0.01_v, share_res_asv_rlt_0.01_f)
junk6$variable <- "No"

# combine results
share_res_byasv_rlt_0.01_2 <- rbind(share_res_byasv_rlt_0.01_2, junk1, junk2, junk3, junk4, junk5, junk6)
share_res_byasv_rlt_0.01_2$group <- factor(share_res_byasv_rlt_0.01_2$group, 
                                         levels=c("Maternal fecal - Infant fecal", "Maternal vaginal - Infant fecal"),
                                labels=c("Sharing with maternal fecal microbiota", "Sharing with maternal vaginal microbiota"))
share_res_byasv_rlt_0.01_2$type <- factor(share_res_byasv_rlt_0.01_2$type, levels=c("Not shared", "Shared"))

```


## Bifidobacterium spp.
### Raw data, 3rd trimester
#### Take a look at all ASVs: total=14, existed in infant=9, shared=5
```{r fig.width=12, fig.height=6}
length(unique(share_res_byasv[share_res_byasv$genus=="Bifidobacterium",]$asv)) 

bifido_dta <- share_res_byasv %>%
  filter(genus=="Bifidobacterium")
bifido_dta$name_md5 <- ifelse(bifido_dta$name_md5=="Bifidobacterium Bifidobacterium bifidum-9175", 
                              "Bifidobacterium bifidum-9175", bifido_dta$name_md5)


bifido_dta[bifido_dta$variable=="Overall",]  %>%
  ggplot(aes(x=variable, y=n_pair, fill=factor(type, levels=c("Not shared", "Shared")))) +
  geom_bar(stat="identity", position="stack", color="black") +    
  facet_grid(name_md5~group, scales = "free_y", switch="y", space = "free_y") +
  ylab("Number of Mother-Infant Pairs") + xlab("Overall") + labs(fill="") +
  theme_bw() +
  theme(axis.title.x = element_text(size=15, face="bold"), axis.title.y = element_text(size=14), axis.ticks.y=element_blank(),
        axis.text.x = element_text(size=13), axis.text.y = element_blank(),
        legend.position="bottom", legend.direction="horizontal", legend.text = element_text(size=13),
        legend.title = element_text(size=13, face="bold"),
        panel.grid = element_blank(),
        strip.text.y.left = element_text(angle = 0, hjust=1), strip.placement = "outside", 
        strip.background.y = element_blank(), strip.text.y = element_text(size=15, face="italic"), 
        strip.text.x = element_text(size=15, color="white"), strip.background.x = element_rect(color=NA, fill="gray"),
        panel.border = element_blank() , axis.line.x = element_line(colour = "black")) +
  coord_flip() +
  scale_fill_manual(values=c("white","#D55E00")) + scale_x_discrete(position="top")
```

Save taxonomy and md5 name for shared Bifido ASVs
```{r}
bifido_name <- bifido_dta[bifido_dta$variable=="Overall" & bifido_dta$type=="Shared", ]
bifido_name <- bifido_name %>%
  group_by(asv) %>%
  mutate(any_share=sum(n_pair))
bifido_name <- bifido_name[bifido_name$any_share>0 & bifido_name$group=="Sharing with maternal fecal microbiota",]
bifido_name <- bifido_name[, c("Taxon", "name_md5", "md5")]
```

#### Make plot
```{r fig.width=40, fig.height=4}
p <- Bifido_Bac_plot(bifido_dta)
p
```
```{r eval=FALSE, include=FALSE}
ggsave(p, filename = "MARCH_3rdtms_bifido_raw_100423.png",
       path =  paste0(path, "Figures"),
       height = 4, width = 40, units = "in", dpi = 300)

```

#### Try heatmap
```{r fig.height=2.5, fig.width=6}
p <- Bifido_Bac_heat(bifido_dta)
p

```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_3rdtms_bifido_raw_100423_v2.png",
       path =  paste0(path, "Figures"),
       height = 2.5, width = 6, units = "in", dpi = 500)
```

```{r eval=FALSE, fig.height=4, fig.width=7, include=FALSE}
# make a figure for legend
asv_dta <- bifido_dta

  # only keep ASVs with evidence of sharing
  junk <- asv_dta[asv_dta$type=="Shared" & asv_dta$variable=="Overall",] %>%
    group_by(name) %>%
    mutate(sum_any_share=sum(n_pair))
  junk <- junk[junk$sum_any_share>0, ]
  
  # calculate pairs of any share overall, which will be used to sort ASVs
  asv_dta <- filter(asv_dta, asv %in% as.list(junk[junk$sum_any_share>0, ]$asv))
  asv_dta <- merge(asv_dta, junk[, c("asv", "sum_any_share", "group")], by=c("asv", "group"))
  
  
  # calculate proportion of shared, which will be added as labels in the figure
  asv_dta <- asv_dta %>%
    group_by(name, group, variable) %>%
    mutate(sum_n=sum(n_pair), prop=n_pair/sum_n*100,
           label=ifelse(sum_n==0, "NA", paste0(round(n_pair/sum_n*100, digits = 1))))

  fig_dta <- asv_dta[asv_dta$variable=="Overall" & asv_dta$type=="Shared",]
  
  fig_overall_1 <-
    ggplot(fig_dta[fig_dta$group=="Sharing with maternal fecal microbiota",], aes(x=1, y=variable, fill=prop)) +
    geom_tile() + coord_fixed() + ylab("Overall") + labs(fill="") +
    geom_text(aes(label = label), color = "black", size = 2) +
    facet_wrap(~ reorder(name_md5, sum_any_share, decreasing=TRUE), ncol = 1, strip.position = "left") + 
    theme_bw() +   
    theme(axis.title = element_blank(), axis.ticks=element_blank(), axis.text = element_blank(), 
          legend.position="bottom", legend.direction = "horizontal" , legend.text = element_text(size=8),
          panel.grid = element_blank(),
          strip.placement = "outside", strip.text.y.left = element_text(angle=0, hjust=1), 
          strip.background.y = element_blank(), strip.text.y = element_text(size=8, face="italic"),
          panel.border = element_blank()) +
    scale_fill_gradient(limits=c(0, 100), low = "white", high = "#0072B2") + scale_y_discrete(position = "right") +
    guides(fill = guide_colourbar(barwidth = 10, barheight = 0.8))

    fig_overall_2 <-
    ggplot(fig_dta[fig_dta$group=="Sharing with maternal vaginal microbiota",], aes(x=1, y=variable, fill=prop)) +
    geom_tile() + coord_fixed() + ylab("Overall") + labs(fill="") +
    geom_text(aes(label = label), color = "black", size = 2) +
    facet_wrap(~ reorder(name_md5, sum_any_share, decreasing=TRUE), ncol = 1, strip.position = "left") + 
    theme_bw() +   
    theme(axis.title.x = element_blank(), axis.ticks=element_blank(), axis.text = element_blank(), 
          axis.title.y = element_text(size=8),
          legend.position="bottom", legend.direction = "horizontal" , legend.text = element_text(size=8),
          panel.grid = element_blank(),
          strip.placement = "outside", strip.text.y.left = element_text(angle=0, hjust=1), 
          strip.background.y = element_blank(), strip.text.y = element_text(size=0, face="italic", color="transparent"),
          panel.border = element_blank()) +
    scale_fill_gradient(limits=c(0, 100), low = "white", high = "#E69F00") + scale_y_discrete(position = "right") +
    guides(fill = guide_colourbar(barwidth = 10, barheight = 0.8))


p <- ggarrange(fig_overall_1, fig_overall_2, nrow=1)
p
    ggsave(p, filename = "MARCH_3rdtms_bifido_raw_100423_v2_legend.png",
       path =  paste0(path, "Figures"),
       height = 4, width = 7, units = "in", dpi = 500)

```


#### Compare how many ASVs within Bacteroides genus are shared and not shared
```{r fig.height=2.2, fig.width=6, message=FALSE}
p <- Bifido_Bac_genus(bifido_dta)
p
```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_3rdtms_bifido_genus_raw_101223.png",
       path =  paste0(path, "Figures"),
       height = 2.2, width = 6, units = "in", dpi = 500)
```


### Adjusted data, 3rd trimester
#### Take a look at all ASVs: total=14, existed in infant=9, shared=5
```{r fig.width=12, fig.height=6}
length(unique(share_res_byasv_rlt_0.01[share_res_byasv_rlt_0.01$genus=="Bifidobacterium",]$asv)) 

bifido_dta <- share_res_byasv_rlt_0.01 %>%
  filter(genus=="Bifidobacterium")
bifido_dta$name_md5 <- ifelse(bifido_dta$name_md5=="Bifidobacterium Bifidobacterium bifidum-9175", 
                              "Bifidobacterium bifidum-9175", bifido_dta$name_md5)


bifido_dta[bifido_dta$variable=="Overall",]  %>%
  ggplot(aes(x=variable, y=n_pair, fill=factor(type, levels=c("Not shared", "Shared")))) +
  geom_bar(stat="identity", position="stack", color="black") +    
  facet_grid(name_md5~group, scales = "free_y", switch="y", space = "free_y") +
  ylab("Number of Mother-Infant Pairs") + xlab("Overall") + labs(fill="") +
  theme_bw() +
  theme(axis.title.x = element_text(size=15, face="bold"), axis.title.y = element_text(size=14), axis.ticks.y=element_blank(),
        axis.text.x = element_text(size=13), axis.text.y = element_blank(),
        legend.position="bottom", legend.direction="horizontal", legend.text = element_text(size=13),
        legend.title = element_text(size=13, face="bold"),
        panel.grid = element_blank(),
        strip.text.y.left = element_text(angle = 0, hjust=1), strip.placement = "outside", 
        strip.background.y = element_blank(), strip.text.y = element_text(size=15, face="italic"), 
        strip.text.x = element_text(size=15, color="white"), strip.background.x = element_rect(color=NA, fill="gray"),
        panel.border = element_blank() , axis.line.x = element_line(colour = "black")) +
  coord_flip() +
  scale_fill_manual(values=c("white","#D55E00")) + scale_x_discrete(position="top")
```

#### Make plot
```{r fig.width=40, fig.height=4}
p <- Bifido_Bac_plot(bifido_dta)
p
```
```{r eval=FALSE, include=FALSE}
ggsave(p, filename = "MARCH_3rdtms_bifido_adj_100423.png",
       path =  paste0(path, "Figures"),
       height = 4, width = 40, units = "in", dpi = 300)

```

#### Try heatmap
```{r fig.height=2.5, fig.width=6}
p <- Bifido_Bac_heat(bifido_dta)
p

```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_3rdtms_bifido_adj_100423_v2.png",
       path =  paste0(path, "Figures"),
       height = 2.5, width = 6, units = "in", dpi = 500)
```

#### Compare how many ASVs within Bacteroides genus are shared and not shared
```{r fig.height=2.2, fig.width=6, message=FALSE}
p <- Bifido_Bac_genus(bifido_dta)
p
```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_3rdtms_bifido_genus_adj_101223.png",
       path =  paste0(path, "Figures"),
       height = 2.2, width = 6, units = "in", dpi = 500)
```


### Raw data, same trimester
#### Take a look at all ASVs: total=14, existed in infant=10, shared=5
```{r fig.width=12, fig.height=6}
length(unique(share_res_byasv_2[share_res_byasv_2$genus=="Bifidobacterium",]$asv)) 

bifido_dta <- share_res_byasv_2 %>%
  filter(genus=="Bifidobacterium")
bifido_dta$name_md5 <- ifelse(bifido_dta$name_md5=="Bifidobacterium Bifidobacterium bifidum-9175", 
                              "Bifidobacterium bifidum-9175", bifido_dta$name_md5)


bifido_dta[bifido_dta$variable=="Overall",]  %>%
  ggplot(aes(x=variable, y=n_pair, fill=factor(type, levels=c("Not shared", "Shared")))) +
  geom_bar(stat="identity", position="stack", color="black") +    
  facet_grid(name_md5~group, scales = "free_y", switch="y", space = "free_y") +
  ylab("Number of Mother-Infant Pairs") + xlab("Overall") + labs(fill="") +
  theme_bw() +
  theme(axis.title.x = element_text(size=15, face="bold"), axis.title.y = element_text(size=14), axis.ticks.y=element_blank(),
        axis.text.x = element_text(size=13), axis.text.y = element_blank(),
        legend.position="bottom", legend.direction="horizontal", legend.text = element_text(size=13),
        legend.title = element_text(size=13, face="bold"),
        panel.grid = element_blank(),
        strip.text.y.left = element_text(angle = 0, hjust=1), strip.placement = "outside", 
        strip.background.y = element_blank(), strip.text.y = element_text(size=15, face="italic"), 
        strip.text.x = element_text(size=15, color="white"), strip.background.x = element_rect(color=NA, fill="gray"),
        panel.border = element_blank() , axis.line.x = element_line(colour = "black")) +
  coord_flip() +
  scale_fill_manual(values=c("white","#D55E00")) + scale_x_discrete(position="top")
```

#### Make plot
```{r fig.width=40, fig.height=4}
p <- Bifido_Bac_plot(bifido_dta)
p
```
```{r eval=FALSE, include=FALSE}
ggsave(p, filename = "MARCH_sametms_bifido_raw_100423.png",
       path =  paste0(path, "Figures"),
       height = 4, width = 40, units = "in", dpi = 300)

```

#### Try heatmap
```{r fig.height=2.5, fig.width=6}
p <- Bifido_Bac_heat(bifido_dta)
p

```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_sametms_bifido_raw_100423_v2.png",
       path =  paste0(path, "Figures"),
       height = 2.5, width = 6, units = "in", dpi = 500)
```

#### Compare how many ASVs within Bacteroides genus are shared and not shared
```{r fig.height=2.2, fig.width=6, message=FALSE}
p <- Bifido_Bac_genus(bifido_dta)
p
```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_sametms_bifido_genus_raw_101223.png",
       path =  paste0(path, "Figures"),
       height = 2.2, width = 6, units = "in", dpi = 500)
```


### Adjusted data, same trimester
#### Take a look at all ASVs: total=14, existed in infant=9, shared=5
```{r fig.width=12, fig.height=6}
length(unique(share_res_byasv_rlt_0.01_2[share_res_byasv_rlt_0.01_2$genus=="Bifidobacterium",]$asv)) 

bifido_dta <- share_res_byasv_rlt_0.01_2 %>%
  filter(genus=="Bifidobacterium")
bifido_dta$name_md5 <- ifelse(bifido_dta$name_md5=="Bifidobacterium Bifidobacterium bifidum-9175", 
                              "Bifidobacterium bifidum-9175", bifido_dta$name_md5)


bifido_dta[bifido_dta$variable=="Overall",]  %>%
  ggplot(aes(x=variable, y=n_pair, fill=factor(type, levels=c("Not shared", "Shared")))) +
  geom_bar(stat="identity", position="stack", color="black") +    
  facet_grid(name_md5~group, scales = "free_y", switch="y", space = "free_y") +
  ylab("Number of Mother-Infant Pairs") + xlab("Overall") + labs(fill="") +
  theme_bw() +
  theme(axis.title.x = element_text(size=15, face="bold"), axis.title.y = element_text(size=14), axis.ticks.y=element_blank(),
        axis.text.x = element_text(size=13), axis.text.y = element_blank(),
        legend.position="bottom", legend.direction="horizontal", legend.text = element_text(size=13),
        legend.title = element_text(size=13, face="bold"),
        panel.grid = element_blank(),
        strip.text.y.left = element_text(angle = 0, hjust=1), strip.placement = "outside", 
        strip.background.y = element_blank(), strip.text.y = element_text(size=15, face="italic"), 
        strip.text.x = element_text(size=15, color="white"), strip.background.x = element_rect(color=NA, fill="gray"),
        panel.border = element_blank() , axis.line.x = element_line(colour = "black")) +
  coord_flip() +
  scale_fill_manual(values=c("white","#D55E00")) + scale_x_discrete(position="top")
```

#### Make plot
```{r fig.width=40, fig.height=4}
p <- Bifido_Bac_plot(bifido_dta)
p
```
```{r eval=FALSE, include=FALSE}
ggsave(p, filename = "MARCH_sametms_bifido_adj_100423.png",
       path =  paste0(path, "Figures"),
       height = 4, width = 40, units = "in", dpi = 300)

```

#### Try heatmap
```{r fig.height=2.5, fig.width=6}
p <- Bifido_Bac_heat(bifido_dta)
p

```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_sametms_bifido_adj_100423_v2.png",
       path =  paste0(path, "Figures"),
       height = 2.5, width = 6, units = "in", dpi = 500)
```

#### Compare how many ASVs within Bacteroides genus are shared and not shared
```{r fig.height=2.2, fig.width=6, message=FALSE}
p <- Bifido_Bac_genus(bifido_dta)
p
```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_sametms_bifido_genus_adj_101223.png",
       path =  paste0(path, "Figures"),
       height = 2.2, width = 6, units = "in", dpi = 500)
```


## Bacteroides spp.
### Raw data, 3rd trimester
#### Take a look at all ASVs: total=122, existed in infant=55, shared=29
```{r fig.width=12, fig.height=30}
length(unique(share_res_byasv[share_res_byasv$genus=="Bacteroides",]$asv)) 

bac_dta <- share_res_byasv %>%
  filter(genus=="Bacteroides")
bac_dta$name_md5 <- ifelse(bac_dta$name_md5=="Bacteroides Bacteroides intestinalis-b87b", "Bacteroides intestinalis-b87b", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides massiliensis-dcc8", "Bacteroides massiliensis-dcc8", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides eggerthii-3342", "Bacteroides eggerthii-3342", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides eggerthii-97c7", "Bacteroides eggerthii-97c7", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides stercoris-9243", "Bacteroides stercoris-9243",
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides coprocola-aab1", "Bacteroides coprocola-aab1",
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides plebeius-e79e", "Bacteroides plebeius-e79e", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides plebeius-fe77", "Bacteroides plebeius-fe77", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides coprophilus-7755", "Bacteroides coprophilus-7755", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides clarus-592f", "Bacteroides clarus-592f", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides stercoris-47be", "Bacteroides stercoris-47be", bac_dta$name_md5)))))))))))


bac_dta[bac_dta$variable=="Overall",]  %>%
  ggplot(aes(x=variable, y=n_pair, fill=factor(type, levels=c("Not shared", "Shared")))) +
  geom_bar(stat="identity", position="stack", color="black") +    
  facet_grid(name_md5~group, scales = "free_y", switch="y", space = "free_y") +
  ylab("Number of Mother-Infant Pairs") + xlab("Overall") + labs(fill="") +
  theme_bw() +
  theme(axis.title.x = element_text(size=15, face="bold"), axis.title.y = element_text(size=14), axis.ticks.y=element_blank(),
        axis.text.x = element_text(size=13), axis.text.y = element_blank(),
        legend.position="bottom", legend.direction="horizontal", legend.text = element_text(size=13),
        legend.title = element_text(size=13, face="bold"),
        panel.grid = element_blank(),
        strip.text.y.left = element_text(angle = 0, hjust=1), strip.placement = "outside", 
        strip.background.y = element_blank(), strip.text.y = element_text(size=15, face="italic"), 
        strip.text.x = element_text(size=15, color="white"), strip.background.x = element_rect(color=NA, fill="gray"),
        panel.border = element_blank() , axis.line.x = element_line(colour = "black")) +
  coord_flip() +
  scale_fill_manual(values=c("white","#D55E00")) + scale_x_discrete(position="top")
```

Save taxonomy and md5 name for shared Bacteroides ASVs
```{r}
bac_name <- bac_dta[bac_dta$variable=="Overall" & bac_dta$type=="Shared", ]
bac_name <- bac_name %>%
  group_by(asv) %>%
  mutate(any_share=sum(n_pair))
bac_name <- bac_name[bac_name$any_share>0 & bac_name$group=="Sharing with maternal fecal microbiota",]
bac_name <- bac_name[, c("Taxon", "name_md5", "md5")]
```


#### Make plot
```{r fig.width=40, fig.height=15}
p <- Bifido_Bac_plot(bac_dta)
p
```
```{r eval=FALSE, include=FALSE}
ggsave(p, filename = "MARCH_3rdtms_bacteroides_raw_100423.png",
       path =  paste0(path, "Figures"),
       height = 15, width = 40, units = "in", dpi = 300)

```

#### Try heatmap
```{r fig.height=15, fig.width=6}
p <- Bifido_Bac_heat(bac_dta)
p
```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_3rdtms_bacteroides_raw_100423_v2.png",
       path =  paste0(path, "Figures"),
       height = 15, width = 6, units = "in", dpi = 500)
```

#### Compare how many ASVs within Bacteroides genus are shared and not shared
```{r fig.height=2.2, fig.width=6, message=FALSE}
p <- Bifido_Bac_genus(bac_dta)
p
```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_3rdtms_bacteroides_genus_raw_101223.png",
       path =  paste0(path, "Figures"),
       height = 2.2, width = 6, units = "in", dpi = 500)
```


### Adjusted data, 3rd trimester
#### Take a look at all ASVs: total=122, existed in infant=55, shared=23
```{r fig.width=12, fig.height=30}
length(unique(share_res_byasv_rlt_0.01[share_res_byasv_rlt_0.01$genus=="Bacteroides",]$asv)) 

bac_dta <- share_res_byasv_rlt_0.01 %>%
  filter(genus=="Bacteroides")
bac_dta$name_md5 <- ifelse(bac_dta$name_md5=="Bacteroides Bacteroides intestinalis-b87b", "Bacteroides intestinalis-b87b", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides massiliensis-dcc8", "Bacteroides massiliensis-dcc8", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides eggerthii-3342", "Bacteroides eggerthii-3342", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides eggerthii-97c7", "Bacteroides eggerthii-97c7", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides stercoris-9243", "Bacteroides stercoris-9243",
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides coprocola-aab1", "Bacteroides coprocola-aab1",
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides plebeius-e79e", "Bacteroides plebeius-e79e", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides plebeius-fe77", "Bacteroides plebeius-fe77", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides coprophilus-7755", "Bacteroides coprophilus-7755", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides clarus-592f", "Bacteroides clarus-592f", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides stercoris-47be", "Bacteroides stercoris-47be", bac_dta$name_md5)))))))))))


bac_dta[bac_dta$variable=="Overall",]  %>%
  ggplot(aes(x=variable, y=n_pair, fill=factor(type, levels=c("Not shared", "Shared")))) +
  geom_bar(stat="identity", position="stack", color="black") +    
  facet_grid(name_md5~group, scales = "free_y", switch="y", space = "free_y") +
  ylab("Number of Mother-Infant Pairs") + xlab("Overall") + labs(fill="") +
  theme_bw() +
  theme(axis.title.x = element_text(size=15, face="bold"), axis.title.y = element_text(size=14), axis.ticks.y=element_blank(),
        axis.text.x = element_text(size=13), axis.text.y = element_blank(),
        legend.position="bottom", legend.direction="horizontal", legend.text = element_text(size=13),
        legend.title = element_text(size=13, face="bold"),
        panel.grid = element_blank(),
        strip.text.y.left = element_text(angle = 0, hjust=1), strip.placement = "outside", 
        strip.background.y = element_blank(), strip.text.y = element_text(size=15, face="italic"), 
        strip.text.x = element_text(size=15, color="white"), strip.background.x = element_rect(color=NA, fill="gray"),
        panel.border = element_blank() , axis.line.x = element_line(colour = "black")) +
  coord_flip() +
  scale_fill_manual(values=c("white","#D55E00")) + scale_x_discrete(position="top")
```

#### Make plot
```{r fig.width=40, fig.height=15}
p <- Bifido_Bac_plot(bac_dta)
p
```
```{r eval=FALSE, include=FALSE}
ggsave(p, filename = "MARCH_3rdtms_bacteroides_adj_100423.png",
       path =  paste0(path, "Figures"),
       height = 15, width = 40, units = "in", dpi = 300)

```


#### Try heatmap
```{r fig.height=11.5, fig.width=6}
p <- Bifido_Bac_heat(bac_dta)
p
```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_3rdtms_bacteroides_adj_100423_v2.png",
       path =  paste0(path, "Figures"),
       height = 11.5, width = 6, units = "in", dpi = 500)
```

#### Compare how many ASVs within Bacteroides genus are shared and not shared
```{r fig.height=2.2, fig.width=6, message=FALSE}
p <- Bifido_Bac_genus(bac_dta)
p
```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_3rdtms_bacteroides_genus_adj_101223.png",
       path =  paste0(path, "Figures"),
       height = 2.2, width = 6, units = "in", dpi = 500)
```


### Raw data, same trimester
#### Take a look at all ASVs: share=29
```{r fig.width=12, fig.height=30}
length(unique(share_res_byasv_2[share_res_byasv_2$genus=="Bacteroides",]$asv)) 

bac_dta <- share_res_byasv_2 %>%
  filter(genus=="Bacteroides")
bac_dta$name_md5 <- ifelse(bac_dta$name_md5=="Bacteroides Bacteroides intestinalis-b87b", "Bacteroides intestinalis-b87b", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides massiliensis-dcc8", "Bacteroides massiliensis-dcc8", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides eggerthii-3342", "Bacteroides eggerthii-3342", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides eggerthii-97c7", "Bacteroides eggerthii-97c7", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides stercoris-9243", "Bacteroides stercoris-9243",
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides coprocola-aab1", "Bacteroides coprocola-aab1",
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides plebeius-e79e", "Bacteroides plebeius-e79e", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides plebeius-fe77", "Bacteroides plebeius-fe77", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides coprophilus-7755", "Bacteroides coprophilus-7755", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides clarus-592f", "Bacteroides clarus-592f", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides stercoris-47be", "Bacteroides stercoris-47be", bac_dta$name_md5)))))))))))


bac_dta[bac_dta$variable=="Overall",]  %>%
  ggplot(aes(x=variable, y=n_pair, fill=factor(type, levels=c("Not shared", "Shared")))) +
  geom_bar(stat="identity", position="stack", color="black") +    
  facet_grid(name_md5~group, scales = "free_y", switch="y", space = "free_y") +
  ylab("Number of Mother-Infant Pairs") + xlab("Overall") + labs(fill="") +
  theme_bw() +
  theme(axis.title.x = element_text(size=15, face="bold"), axis.title.y = element_text(size=14), axis.ticks.y=element_blank(),
        axis.text.x = element_text(size=13), axis.text.y = element_blank(),
        legend.position="bottom", legend.direction="horizontal", legend.text = element_text(size=13),
        legend.title = element_text(size=13, face="bold"),
        panel.grid = element_blank(),
        strip.text.y.left = element_text(angle = 0, hjust=1), strip.placement = "outside", 
        strip.background.y = element_blank(), strip.text.y = element_text(size=15, face="italic"), 
        strip.text.x = element_text(size=15, color="white"), strip.background.x = element_rect(color=NA, fill="gray"),
        panel.border = element_blank() , axis.line.x = element_line(colour = "black")) +
  coord_flip() +
  scale_fill_manual(values=c("white","#D55E00")) + scale_x_discrete(position="top")
```

#### Make plot
```{r fig.width=40, fig.height=15}
p <- Bifido_Bac_plot(bac_dta)
p
```
```{r eval=FALSE, include=FALSE}
ggsave(p, filename = "MARCH_sametms_bacteroides_raw_100423.png",
       path =  paste0(path, "Figures"),
       height = 15, width = 40, units = "in", dpi = 300)

```


#### Try heatmap
```{r fig.height=15, fig.width=6}
p <- Bifido_Bac_heat(bac_dta)
p
```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_sametms_bacteroides_raw_100423_v2.png",
       path =  paste0(path, "Figures"),
       height = 15, width = 6, units = "in", dpi = 500)
```

#### Compare how many ASVs within Bacteroides genus are shared and not shared
```{r fig.height=2.2, fig.width=6, message=FALSE}
p <- Bifido_Bac_genus(bac_dta)
p
```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_sametms_bacteroides_genus_raw_101223.png",
       path =  paste0(path, "Figures"),
       height = 2.2, width = 6, units = "in", dpi = 500)
```


### Adjusted data, same trimester
#### Take a look at all ASVs: share=23
```{r fig.width=12, fig.height=30}
length(unique(share_res_byasv_rlt_0.01_2[share_res_byasv_rlt_0.01_2$genus=="Bacteroides",]$asv)) 

bac_dta <- share_res_byasv_rlt_0.01_2 %>%
  filter(genus=="Bacteroides")
bac_dta$name_md5 <- ifelse(bac_dta$name_md5=="Bacteroides Bacteroides intestinalis-b87b", "Bacteroides intestinalis-b87b", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides massiliensis-dcc8", "Bacteroides massiliensis-dcc8", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides eggerthii-3342", "Bacteroides eggerthii-3342", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides eggerthii-97c7", "Bacteroides eggerthii-97c7", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides stercoris-9243", "Bacteroides stercoris-9243",
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides coprocola-aab1", "Bacteroides coprocola-aab1",
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides plebeius-e79e", "Bacteroides plebeius-e79e", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides plebeius-fe77", "Bacteroides plebeius-fe77", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides coprophilus-7755", "Bacteroides coprophilus-7755", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides clarus-592f", "Bacteroides clarus-592f", 
                           ifelse(bac_dta$name_md5=="Bacteroides Bacteroides stercoris-47be", "Bacteroides stercoris-47be", bac_dta$name_md5)))))))))))


bac_dta[bac_dta$variable=="Overall",]  %>%
  ggplot(aes(x=variable, y=n_pair, fill=factor(type, levels=c("Not shared", "Shared")))) +
  geom_bar(stat="identity", position="stack", color="black") +    
  facet_grid(name_md5~group, scales = "free_y", switch="y", space = "free_y") +
  ylab("Number of Mother-Infant Pairs") + xlab("Overall") + labs(fill="") +
  theme_bw() +
  theme(axis.title.x = element_text(size=15, face="bold"), axis.title.y = element_text(size=14), axis.ticks.y=element_blank(),
        axis.text.x = element_text(size=13), axis.text.y = element_blank(),
        legend.position="bottom", legend.direction="horizontal", legend.text = element_text(size=13),
        legend.title = element_text(size=13, face="bold"),
        panel.grid = element_blank(),
        strip.text.y.left = element_text(angle = 0, hjust=1), strip.placement = "outside", 
        strip.background.y = element_blank(), strip.text.y = element_text(size=15, face="italic"), 
        strip.text.x = element_text(size=15, color="white"), strip.background.x = element_rect(color=NA, fill="gray"),
        panel.border = element_blank() , axis.line.x = element_line(colour = "black")) +
  coord_flip() +
  scale_fill_manual(values=c("white","#D55E00")) + scale_x_discrete(position="top")
```

#### Make plot
```{r fig.width=40, fig.height=12}
p <- Bifido_Bac_plot(bac_dta)
p
```
```{r eval=FALSE, include=FALSE}
ggsave(p, filename = "MARCH_sametms_bacteroides_adj_100423.png",
       path =  paste0(path, "Figures"),
       height = 12, width = 40, units = "in", dpi = 300)

```


#### Try heatmap
```{r fig.height=11.5, fig.width=6}
p <- Bifido_Bac_heat(bac_dta)
p
```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_sametms_bacteroides_adj_100423_v2.png",
       path =  paste0(path, "Figures"),
       height = 11.5, width = 6, units = "in", dpi = 500)
```

#### Compare how many ASVs within Bacteroides genus are shared and not shared
```{r fig.height=2.2, fig.width=6, message=FALSE}
p <- Bifido_Bac_genus(bac_dta)
p
```
```{r eval=FALSE, include=FALSE}
    ggsave(p, filename = "MARCH_sametms_bacteroides_genus_adj_101223.png",
       path =  paste0(path, "Figures"),
       height = 2.2, width = 6, units = "in", dpi = 500)
```


## Save taxonomy and md5 name
```{r}
save_name <- rbind(bifido_name, bac_name)
save_name <- save_name[, c(2, 3, 1)]
colnames(save_name) <- c("ASV Name", "MD5 Hash", "Taxonomy")

write.csv(save_name, paste0(path, "Text files/", "MARCH_shared_ASV_info_100523.csv"), row.names = FALSE)
```





