---
title: "FEAST Analysis for MARCH 16S Microbiome (12902) using 54 dyads"
author: "Tiange Liu"
date: "2023-09-08"
output: html_document
---
Last updated on 9/8/2023

```{r set up, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r message=FALSE}
library(dplyr)
library(ggplot2)
library(ggpubr)
library(qiime2R)
library(FEAST)
library(data.table)
library(table1)
library(tidyr)
library(readr)
library(tidyverse)
```

# Read in participant data, explore, and modify
```{r}
march <- read.csv("Z:/DAC_AnalysisProjects/10_Analysis_Proposals/EC0556A/DATA/DERIVED/12902/MARCH_paired_54.csv")
  # these 54 dyads include additional 4 infants: set 6, 5-6 months when samples were collected

table(march$m16s_part_type)
march$participant_type <- as.factor(march$m16s_part_type)
march$participant_type <- factor(march$participant_type, levels=c("1", "2"), labels=c("biological mother", "child"))

table(march$m16s_spec_type)
march$body_site <- as.factor(march$m16s_spec_type)
march$body_site <- factor(march$m16s_spec_type, levels=c("1", "7"), labels=c("stool", "vaginal"))

march$paired_ID <- substr(march$ParticipantID, 1, 9)
```

# Read in covariates 
```{r}
master_dt <- read.csv("Z:/DAC_AnalysisProjects/10_Analysis_Proposals/EC0556A/DATA/DERIVED/OIF_master_SQL_072823_v3.csv")
```

   
# FEAST
## Prepare ASV table for FEAST
```{r}
asv_table <- read_tsv("Z:/DAC_AnalysisProjects/10_Analysis_Proposals/EC0556A/DATA/DERIVED/20230518/01.CohortMerge/03.cohort_asv_data/12902/12902_asv_table_classic.tsv")
  # this is newer asv table Justine created in May 2023

#asv_table$type 
#asv_table$data[1:10, 1:10]
asv_table <- as.data.frame(t(asv_table)) 
  # rows are samples, columns are taxa
colnames(asv_table) <- asv_table[1, ]
  # assign colnames as ASV id
asv_table <- asv_table[-(1:2), ] # remove unnecessary rows

# only keep paired samples
asv_table_paired <- asv_table %>% filter(row.names(asv_table) %in% as.list(march$sample.id))
row_names <- row.names(asv_table_paired)

str(asv_table_paired[1:5, 1:5]) # counts are not integer now
asv_table_paired <- as.data.frame(lapply(asv_table_paired, as.integer)) 
  # transform counts to integer, which is required by FEAST

row.names(asv_table_paired) <- row_names

asv_table_paired <- asv_table_paired[order(row.names(asv_table_paired)), ]
asv_table_paired <- as.matrix(asv_table_paired)
  # this will be the input ASV table for FEAST
```
   
## Prepare input meta table for FEAST
```{r}
meta_dt <- march[, c("sample.id", "paired_ID", "body_site", "participant_type")]
meta_dt <- meta_dt[order(meta_dt$paired_ID), ] 

meta_dt$SourceSink <- ifelse(meta_dt$participant_type=="child", "Sink", "Source")
  # child microbiome = sink, maternal microbiome = source

# per requirement of FEAST, each pair should have same id
meta_dt$id <- 1
n <- 1
for(i in 2:nrow(meta_dt)){
  if(meta_dt$SourceSink[i]=="Sink"){
    meta_dt$id[i] <- n+1
    n <- n+1
  }else{
    meta_dt$id[i] <- meta_dt$id[i-1]
  }
}

meta_dt$Env <- paste(meta_dt$participant_type, meta_dt$body_site, meta_dt$id)

row.names(meta_dt) <- meta_dt$sample.id
meta_dt <- meta_dt[, c("sample.id", "Env", "SourceSink", "id")]
meta_dt$id <- as.integer(meta_dt$id) # integer is required by FEAST
meta_dt <- meta_dt[order(meta_dt$sample.id), ]
meta_dt <- as.data.frame(meta_dt)
meta_dt <- meta_dt[, c("Env", "SourceSink", "id")]
  # this will be the input metadata for FEAST

# check if rownames match (required by FEAST)
common.sample.ids <- intersect(rownames(meta_dt), rownames(asv_table_paired))
length(common.sample.ids) # 162
```
   

## Run FEAST (own mother)
```{r eval=FALSE, warning=FALSE, include=FALSE}
set.seed(60423)

FEAST_output <- FEAST(C = asv_table_paired, metadata = meta_dt, different_sources_flag = 1, 
                      dir_path="Z:/DAC_AnalysisProjects/10_Analysis_Proposals/EC0556A/USERS/TiangeLiu/OUTPUT/Results",   
                      outfile="feast_12902_090823_54dyads")
  # default: 1000 EM iteration; 
  # rarefied at minimal sequencing depth within each group of sink and corresponding source
```
   

# Modify FEAST output for visulization (own mother)
```{r}
output <- read.table("Z:/DAC_AnalysisProjects/10_Analysis_Proposals/EC0556A/USERS/TiangeLiu/OUTPUT/Results/feast_12902_090823_54dyads_source_contributions_matrix.txt")

# select columns containing results when mother stool is the source
output_stool <- output %>% select(contains("stool"))

# select columns containing results when mother vagina is the source
output_vagina <- output %>% select(contains("vaginal"))  

# generate a new output 
output_full <- as.data.frame(row.names(output))
output_full$stool <- rowSums(output_stool, na.rm = TRUE) * ifelse(rowSums(is.na(output_stool)) == ncol(output_stool), NA, 1) * 100
output_full$vaginal <- rowSums(output_vagina, na.rm = TRUE) * ifelse(rowSums(is.na(output_vagina)) == ncol(output_vagina), NA, 1) * 100
  # keep it as NA if no data from the corresponding mother and body site

output_full$unknown <- output$Unknown * 100
  # multiply all estimated contributions by 100 to get the proportion (%)

colnames(output_full)[1] <- "sample.id"
output_full$sample.id <- substr(output_full$sample.id, 1,30)

output_full <- filter(output_full, !is.na(stool) | !is.na(vaginal))
output_full$ParticipantID <- substr(output_full$sample.id, 7, 17)


# add covariates
output_full <- merge(output_full, master_dt[, -c(1)], by.x="ParticipantID", by.y="ParticipantID.x", all=FALSE)

output_full$ParticipantID.y <- NULL

# add age
output_full <- merge(output_full, march[, c("sample.id", "m16s_age_yr")], by="sample.id")


# re-label some variables
output_full$birth_delivm <- as.factor(output_full$birth_delivm)
output_full$birth_delivm <- factor(output_full$birth_delivm, levels=c("1","2"), labels=c("Vaginal delivery", "Cesarean delivery"))
table(output_full$birth_delivm)

output_full$med_antib_pn <- as.factor(output_full$med_antib_pn)
output_full$med_antib_pn <- factor(output_full$med_antib_pn, levels=c("0", "1"), labels=c("No antibiotics", "Took antibiotics"))
table(output_full$med_antib_pn)

table(output_full$birth_delivm, output_full$med_antib_pn)
```

## Create one output file for infants with both maternal sources available
````{r}
# generate another long dataset for infants with both maternal stool and vaginal samples
output_full_3 <- filter(output_full, !is.na(stool) & !is.na(vaginal))
output_full_3 <- gather(output_full_3, type, prop, stool:unknown, factor_key=TRUE)
output_full_3$type <- factor(output_full_3$type, levels=c("stool", "vaginal", "unknown"),
                                labels=c("Stool", "Vagina", "Other"))
length(unique(output_full_3$sample.id))
  # 54

# save output for these 54
write.csv(output_full_3, "Z:/DAC_AnalysisProjects/10_Analysis_Proposals/EC0556A/USERS/TiangeLiu/OUTPUT/Results/feast_12902_091123_54dyads.csv", row.names=FALSE)

```
